azuread/data_user.go

```diff
@@ -0,0 +1,59 @@
+package azuread
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/graphrbac/1.6/graphrbac"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azuread/azuread/helpers/ar"
+	"github.com/terraform-providers/terraform-provider-azuread/azuread/helpers/validate"
+)
+
+func dataSourceUser() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceUserRead,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"user_principal_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"mail": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
```

Comment:
Could we add `display_name` and `mail_nickname` (if different then mail) here?

Reply:
Given that they are exposed by the resource, we should likely have them here. Generally we have parity between the two.

Reply:
Thanks for the feedback. 
Code updated accordingly.